torch/cuda/__init__.py

```diff
@@ -531,7 +534,7 @@ def _raw_device_count_nvml() -> int:
 def _device_count_nvml() -> int:
     """Return number of devices as reported by NVML taking CUDA_VISIBLE_DEVICES into account.
     Negative value is returned if NVML discovery or initialization has failed."""
-    visible_devices = _parse_visible_devices()
+    visible_devices = set(_parse_visible_devices())
```

Comment:
```suggestion
    visible_devices = _parse_visible_devices()
```

`set.intersection(self, Iterable)` accepts `list`.